great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -113,9 +113,7 @@ def batch_request(self) -> Optional[Union[BatchRequest, RuntimeBatchRequest, dic
         return self._batch_request
 
     @batch_request.setter
-    def batch_request(
-        self, batch_request: Union[BatchRequest, RuntimeBatchRequest, dict]
-    ) -> None:
+    def batch_request(self, batch_request: dict) -> None:
```

Comment:
@cdkini The issue I see here is that a `batch_request` can be converted to dictionary and still have a DataFrame.  So I was wondering if it is possible to avoid these setters on the builder objects.